public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -103,10 +104,10 @@ export default class CloudMonitoringDatasource extends DataSourceWithBackend<
       .toPromise();
   }
 
-  applyTemplateVariables(
-    { metricQuery, refId, queryType, sloQuery }: CloudMonitoringQuery,
-    scopedVars: ScopedVars
-  ): Record<string, any> {
+  applyTemplateVariables(query: CloudMonitoringQuery, scopedVars: ScopedVars): Record<string, any> {
+    const { refId, queryType, sloQuery } = query;
+    // Query won't have the key metricQuery in v6.x, just the plain object
+    const metricQuery = query.metricQuery ? query.metricQuery : (query as MetricQuery);
```

Comment:
I think you should be able to call `this.migrateQuery(query)` instead. Then we don't have the migration code at different places.